src/eventListeners/onImageRenderedBrushEventHandler.test.js

```diff
@@ -0,0 +1,510 @@
+      }
+
+      if (viewport.vflip) {
+        scale(1, -1);
+      }
+
+      // Move back from center of image
+      translate(-width / 2, -height / 2);
+
+      const x = imageCoord.x;
+      const y = imageCoord.y;
+
+      const px = x * m[0] + y * m[2] + m[4];
+      const py = x * m[1] + y * m[3] + m[5];
+
+      return {
+        x: px,
+        y: py,
+      };
+    },
+  },
+}));
+
+let eventData;
+let evt;
+let labelmap3D;
+let labelmap2D;
+let lineWidth;
+
+const mockGetEnabledElement = element => ({
+  element,
+  viewport: eventData.viewport,
+  image: eventData.image,
+  canvas: eventData.canvasContext.canvas,
+});
+
+function resetEvents() {
+  const width = 256;
+  const length = width * width;
+  const currentImageIdIndex = 0;
+
+  const canvasScale = 1.0;
+
+  lineWidth = 1;
+
+  eventData = {
+    element: null,
+    image: {
+      width: 256,
+      height: 256,
+    },
+    viewport: {
+      rotation: 0,
+      scale: canvasScale,
+      translation: { x: 0, y: 0 },
+      hflip: false,
+      vflip: false,
+      displayedArea: {
+        brhc: { x: 256, y: 256 },
+        tlhc: { x: 1, y: 1 },
+      },
+    },
+    canvasContext: {
+      canvas: {
+        width: canvasScale * width,
+        height: canvasScale * width,
+      },
+    },
+  };
+
+  evt = {
+    detail: eventData,
+  };
+
+  labelmap3D = {
+    buffer: new ArrayBuffer(length * 2),
+    labelmaps2D: [],
+    metadata: [],
+    activeSegmentIndex: 0,
+    segmentsVisible: [],
+  };
+
+  labelmap2D = {
+    pixelData: new Uint16Array(labelmap3D.buffer, 0, length),
+    segmentsOnLabelmap: [0, 1, 2],
+    invalidated: true,
+  };
+
+  const pixelData = labelmap2D.pixelData;
+  const cols = eventData.image.width;
+
+  // Add segment 1 as an L shape, so should have 1 interior corner.
+  pixelData[64 * cols + 64] = 1;
+  pixelData[65 * cols + 64] = 1;
+  pixelData[65 * cols + 65] = 1;
+
+  // Add segment 2 as a rectangle.
+  for (let x = 201; x <= 210; x++) {
+    pixelData[200 * cols + x] = 2;
+    pixelData[201 * cols + x] = 2;
+  }
+
+  labelmap3D.labelmaps2D[currentImageIdIndex] = labelmap2D;
+}
+
+function setCanvasTransform(options = {}) {
+  const viewport = eventData.viewport;
+
+  if (options.scale) {
+    const { scale } = options;
+    const { width, height } = eventData.image;
+
+    eventData.viewport.scale = scale;
+
+    eventData.canvasContext.canvas = {
+      width: scale * width,
+      height: scale * height,
+    };
+  }
+
+  if (options.rotation) {
+    viewport.rotation = options.rotation;
+  }
+
+  if (options.translation) {
+    viewport.translation = options.translation;
+  }
+
+  if (options.hflip) {
+    viewport.hflip = options.hflip;
+  }
+
+  if (options.vflip) {
+    viewport.vflip = options.vflip;
+  }
+}
+
+describe('onImageRenderedBrushEventHandler.js', () => {
+  beforeEach(() => {
+    resetEvents();
+  });
+
+  describe('Initialization', () => {
+    it('should have a test image of 256 x 256 pixels.', () => {
```

Comment:
These tests are for our mock implementation?

Reply:
Yeah, eventually some e2e tests for this stuff would be better.